app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/AddressLookupRoutingController.scala

```diff
@@ -0,0 +1,79 @@
+
+import javax.inject.Inject
+import play.api.i18n.Lang
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.SelfEmploymentDataKeys.addressIdKey
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.config.AppConfig
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.httpparser.addresslookup.GetAddressLookupDetailsHttpParser.InvalidJson
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.httpparser.addresslookup.PostAddressLookupHttpParser.PostAddressLookupSuccessResponse
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.httpparser.addresslookup._
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.{AddressLookupConnector, IncomeTaxSubscriptionConnector}
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.models.BusinessAddressModel
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.services.AuthService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+
+import scala.concurrent.ExecutionContext
+
+class AddressLookupRoutingController @Inject()(mcc: MessagesControllerComponents,
+                                               authService: AuthService,
+                                               addressLookupConnector: AddressLookupConnector,
+                                               incomeTaxSubscriptionConnector: IncomeTaxSubscriptionConnector
+                                              )(implicit val ec: ExecutionContext, val appConfig: AppConfig) extends FrontendController(mcc) {
+
+  def initialiseAddressLookupJourney(): Action[AnyContent] = Action.async { implicit request =>
+    authService.authorised() {
+      implicit val lang: Lang = mcc.messagesApi.preferred(request).lang
+      val continueUrl = appConfig.incomeTaxSubscriptionSelfEmployedFrontendBaseUrl + routes.AddressLookupRoutingController.addressLookupRedirect(None).url
+      addressLookupConnector.initialiseAddressLookup(continueUrl) map (
+        response =>
+          response match {
+            case Right(PostAddressLookupSuccessResponse(Some(location))) => Redirect(location)
+            case Left(PostAddressLookupHttpParser.UnexpectedStatusFailure(status)) => throw new InternalServerException(
+              s"[AddressLookupRoutingController][initialiseAddressLookupJourney] - Unexpected response, status: $status")
+          }
+        )
+    }
+  }
+
+  def addressLookupRedirect(id: Option[String]): Action[AnyContent] = Action.async { implicit request =>
+    authService.authorised() {
+      if (!id.isDefined) {
+        throw new InternalServerException(
+          s"[AddressLookupRoutingController][addressLookupRedirect] - Id not returned from address service")
+      } else {
+        addressLookupConnector.getAddressDetails(id.get) flatMap {
```

Comment:
This connector method (not your controller code) is broken because the model (BusinessAddressModel) does not have a redirect URL. I don't know where else to comment about this being broken.

Reply:
Done